src/execution/service.ts

```diff
@@ -746,6 +746,8 @@ export class ServiceScriptExecution extends BaseExecutionWithCommand<ServiceScri
       case 'unstarted':
       case 'depsStarting': {
         this._state = {id: 'stopped'};
+        this._terminated.resolve({ok: true, value: undefined});
```

Comment:
Why don't these need to be resolved above?

If these are pre-conditions for entering the stopped state, it'd be nice if they could be checked / enforced there. That would require abstracting over entering states so that you don't just do:

```ts
this._state = {id: 'stopped'};
```

but

```ts
this._enterStoppedState();
```

etc

Reply:
The above cases are going to the `stopped` state. We don't enter `stopped` until the child is killed (or never starts).

I do like the `_enterStoppedState` idea though, since there is another spot above where we enter that state. Also renamed `_fail` to `_enterFailedState` to match.

Reply:
(Edit: typo)